mesonbuild/mcompile.py

```diff
@@ -27,6 +28,20 @@
 if T.TYPE_CHECKING:
     import argparse
 
+def get_backend_from_introspect(builddir: Path) -> str:
+    """
+    Gets `backend` option value from introspection data
+    """
+    path_to_intro = builddir / 'meson-info' / 'intro-buildoptions.json'
```

Comment:
I think you should get it from coredata file instead. Should be faster to pickle that file than parsing json.

Reply:
Fixed